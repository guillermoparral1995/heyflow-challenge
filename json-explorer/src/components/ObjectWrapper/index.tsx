import { PropsWithChildren } from "react";

const ObjectWrapper: React.FC<PropsWithChildren> = ({ children }) => {
  return (
    <>
      <span>{"{"}</span>
      <div className="pl-4">
        <ul>{children}</ul>
      </div>
      <span>{"}"}</span>
    </>
  );
};

export default ObjectWrapper;
