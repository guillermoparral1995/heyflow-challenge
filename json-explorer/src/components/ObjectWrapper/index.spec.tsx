import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import ObjectWrapper from ".";

describe("ObjectWrapper", () => {
  it("should render correctly", () => {
    const { container } = render(<ObjectWrapper />);
    expect(container).toMatchSnapshot();
  });

  it("should render children correctly", () => {
    const children = (
      <>
        <li>Item 1</li>
        <li>Item 2</li>
      </>
    );
    const { getByText } = render(<ObjectWrapper>{children}</ObjectWrapper>);
    expect(getByText("Item 1")).toBeInTheDocument();
    expect(getByText("Item 2")).toBeInTheDocument();
  });

  it("should match snapshot with children", () => {
    const children = (
      <>
        <li>Item 1</li>
        <li>Item 2</li>
      </>
    );
    const { container } = render(<ObjectWrapper>{children}</ObjectWrapper>);
    expect(container).toMatchSnapshot();
  });
});
