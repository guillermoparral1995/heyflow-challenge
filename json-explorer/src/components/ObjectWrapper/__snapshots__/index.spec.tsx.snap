// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectWrapper should match snapshot with children 1`] = `
<div>
  <span>
    {
  </span>
  <div
    class="pl-4"
  >
    <ul>
      <li>
        Item 1
      </li>
      <li>
        Item 2
      </li>
    </ul>
  </div>
  <span>
    }
  </span>
</div>
`;

exports[`ObjectWrapper should render correctly 1`] = `
<div>
  <span>
    {
  </span>
  <div
    class="pl-4"
  >
    <ul />
  </div>
  <span>
    }
  </span>
</div>
`;
